specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-06-01-preview/containerregistry_build.json

```diff
@@ -0,0 +1,2643 @@
+            "schema": {
+              "$ref": "#/definitions/Task"
+            }
+          },
+          "201": {
+            "description": "The request was successfully accepted; the operation will complete asynchronously. The provisioning state of the resource should indicate the current state of the resource.",
+            "schema": {
+              "$ref": "#/definitions/Task"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed. If the registry doesn't exist, 404 (Not found) is returned. If any of the input parameters is wrong, 400(Bad Request) is returned."
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Tasks_Update": {
+            "$ref": "./examples/TasksUpdate.json"
+          },
+          "Tasks_Update_WithKeyVaultCustomCredentials": {
+            "$ref": "./examples/ManagedIdentity/TasksUpdate_WithKeyVaultCustomCredentials.json"
+          },
+          "Tasks_Update_WithMSICustomCredentials": {
+            "$ref": "./examples/ManagedIdentity/TasksUpdate_WithMSICustomCredentials.json"
+          },
+          "Tasks_Update_WithOpaqueCustomCredentials": {
+            "$ref": "./examples/TasksUpdate_WithOpaqueCustomCredentials.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerRegistry/registries/{registryName}/tasks/{taskName}/listDetails": {
+      "post": {
+        "tags": [
+          "Tasks"
+        ],
+        "description": "Returns a task with extended information that includes all secrets.",
+        "operationId": "Tasks_GetDetails",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/RegistryNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/TaskNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request was successful; the request was well-formed and received properly.",
+            "schema": {
+              "$ref": "#/definitions/Task"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed. If the registry doesn't exist, 404 (Not found) is returned."
+          }
+        },
+        "x-ms-examples": {
+          "Tasks_GetDetails": {
+            "$ref": "./examples/TasksGetDetails.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "RunRequest": {
+      "description": "The request parameters for scheduling a run.",
+      "type": "object",
```

Comment:
`"required": [ "type"]` should be present over here since the property is defined in this model and not in the models that have an allOf on this model. Please remove `"type"` from the required property list of all the child model definitions.

Reply:
hmm, for some reason our swagger generator is putting it in the subclasses, not sure why cc @jaysterp

Reply:
@amarzavery We put `"type"`  as the `[Required]` abstract property on the base class. All the implementation classes don't have a "[Required]" attribute on the `type`. We generate our Swagger using Swashbuckle. And it seems to have added `"required": [ "type" ]` in the swagger of base class + sub classes. 
It seems like this is a limitation of Swashbuckle. What do you suggest we do here? 

Also, note that all tests have passed now.